plugins/source/aws/resources/plugin/client.go

```diff
@@ -0,0 +1,55 @@
+package plugin
+
+import (
+	"context"
+	"encoding/json"
+
+	"github.com/cloudquery/cloudquery/plugins/source/aws/client"
+	"github.com/cloudquery/plugin-sdk/v4/message"
+	"github.com/cloudquery/plugin-sdk/v4/plugin"
+	"github.com/cloudquery/plugin-sdk/v4/scheduler"
+	"github.com/cloudquery/plugin-sdk/v4/schema"
+	"github.com/rs/zerolog"
+)
+
+type Client struct {
+	plugin.UnimplementedDestination
+	scheduler *scheduler.Scheduler
+	client    schema.ClientMeta
+}
+
+func New(ctx context.Context, logger zerolog.Logger, specBytes []byte) (plugin.Client, error) {
+	var spec client.Spec
+	c := &Client{}
+	var err error
+	if err := json.Unmarshal(specBytes, &spec); err != nil {
+		return nil, err
+	}
+	if err := spec.Validate(); err != nil {
+		return nil, err
+	}
+	c.client, err = client.Configure(ctx, logger, spec)
+	if err != nil {
+		return nil, err
+	}
+
+	c.scheduler = scheduler.NewScheduler(
+		c.client,
+		scheduler.WithConcurrency(spec.Concurrency),
+		scheduler.WithLogger(logger),
+		scheduler.WithDeterministicCQId(true),
```

Comment:
We probably don't want to _always_ set the CQ ID deterministically, or do we?

Reply:
Is there a good reason to not make the default `true` and enable people to disable the functionality? If we don't have (or hear) a good reason to ever want a random CQID in the future we could remove the option

Reply:
I don't think we should change a default in this PR as already many changes.